src/main/java/com/theironyard/Controllers/IronQuotesController.java

```diff
@@ -0,0 +1,244 @@
+
+    @RequestMapping(path = "/quotes/{quoteId}/tags/{tagId}", method = RequestMethod.DELETE)
+    public Quote deleteTag(@PathVariable int tagId, @PathVariable int quoteId,
+                           @RequestHeader(value = "Authorization") String auth) throws Exception {
+
+        User user = getUserFromAuth(auth);
+
+        Tag tag = tagRepository.getOne(tagId);
+        Quote quote = quoteRepository.getOne(quoteId);
+        if (tag == null) {
+            throw new Exception("Tag doesn't exist");
+        } else {
+            quote.deleteTag(tag);
+            quoteRepository.save(quote);
+        }
+
+        return quote;
+    }
+
+
+    //*********LIST-PATHS*********//
+    @RequestMapping(path = "/quotes", method = RequestMethod.GET)
+    public List<Quote> getQuotes() {
+        return quoteRepository.findAll();
+    }
+
+    @RequestMapping(path = "/tags", method = RequestMethod.GET)
+    public List<Tag> getTags() {
+        return tagRepository.findAll();
+    }
+
+
+    //*********USER-PATHS & METHODS*********//
+
+    @RequestMapping(path = "/token", method = RequestMethod.POST)
+    public Map getToken(@RequestBody UserCommand command) throws Exception {
+        User user = checkUser(command);
+
+        Map<String, String> tokenMap = new HashMap<>();
+        tokenMap.put("token", user.getToken());
+        return tokenMap;
+    }
+
+    @RequestMapping(path = "/token/regenerate", method = RequestMethod.PUT)
+    public Map regenerateToken(@RequestBody UserCommand command) throws Exception {
+        User user = checkUser(command);
+        String token = user.regenerate();
+        Map<String, String> tokenMap = new HashMap<>();
+        tokenMap.put("token", token);
+        return tokenMap;
+    }
+
+    @RequestMapping(path = "/users", method = RequestMethod.POST)
+    public User login(@RequestBody UserCommand command) throws Exception {
+        User user = userRepository.findFirstByName(command.getName());
+
+        if (user == null) {
+            user = new User(command.getName(), PasswordStorage.createHash(command.getPassword()));
+            userRepository.save(user);
+        } else if (!PasswordStorage.verifyPassword(command.getPassword(), user.getPassword())) {
+            throw new LoginFailedException();
+        }
+
+        return user;
+    }
+
+    public User checkUser(UserCommand command) throws Exception {
+        User user = userRepository.findFirstByName(command.getName());
+        if (user == null) {
+            throw new UserNotFoundException();
+        }
+
+        if (!PasswordStorage.verifyPassword(command.getPassword(), user.getPassword())) {
+            throw new LoginFailedException();
+        }
+
+        if (!user.isTokenValid()) {
+            throw new TokenExpiredException();
+        }
+
+        return user;
+    }
+
+    public User getUserFromAuth(String auth) {
+        User user = userRepository.findFirstByToken(auth.split(" ")[1]);
+        if (!user.isTokenValid()) {
```

Comment:
This will throw a null pointer in the event that the token fails to get a user.  You need to check the user then check the validity of the token